src/mstch.cpp

```diff
@@ -478,18 +479,48 @@ ::mstch::node CXXRecord::methods()
     if (method_vector.size() > 0)
         method_vector.back()->setLast(true);
 
-    // Copy each template into the mstch template array.
-    for (auto method_template : method_vector)
-        method_templates.push_back(method_template);
+    // Collect all the non-static method names
+    std::unordered_set<std::string> non_static_method_names;
+    for (const std::shared_ptr<Method> &method : method_vector)
+    {
+        const bool is_static = boost::get<bool>(method->at("is_static"));
+        if (is_static)
+            continue;
+
+        non_static_method_names.insert(method->nameAsString());
```

Comment:
Oops, wait, do we need this here too?
```suggestion
        non_static_method_names.insert(method->nameAsString());
        method_templates.push_back(method);
```